resources/helm/dask-gateway/templates/gateway/middleware.yaml

```
@@ -0,0 +1,23 @@
+apiVersion: traefik.containo.us/v1alpha1
+kind: Middleware
+metadata:
+  name: {{ include "dask-gateway.fullname" . | printf "clusters-prefix-%s" | trunc 63 | trimSuffix "-" }}
+  labels:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+spec:
+  stripPrefixRegex:
+    regex:
+      - '{{ .Values.gateway.prefix | trimSuffix "/" }}/clusters/[a-zA-Z0-9.-]+/'
+{{- if ne .Values.gateway.prefix "/" }}
+---
+apiVersion: traefik.containo.us/v1alpha1
+kind: Middleware
+metadata:
+  name: {{ include "dask-gateway.fullname" . | printf "api-prefix-%s" | trunc 63 | trimSuffix "-" }}
+  labels:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+spec:
+  stripPrefixRegex:
+    regex:
+      - '{{ .Values.gateway.prefix | trimSuffix "/" }}/'
```

Similarly as above, I consider:

```suggestion
      - '{{ .Values.gateway.prefix | trimSuffix "/" }}'
```